wagtail/models/__init__.py

```diff
@@ -2136,15 +2136,16 @@ def get_url_parts(self, request=None):
         if not possible_sites:
             return None
 
+        unique_site_ids = {values[0] for values in possible_sites}
```

Comment:
Question: Is there a reason a site would appear multiple times?

Reply:
@RealOrangeOne Yes. When `WAGTAIL_I18N_ENABLED=True`, there will be a language-specific route for each site. See https://github.com/wagtail/wagtail/blob/main/wagtail/models/sites.py#L206-L265